scifight/tournament_specific.py

```diff
@@ -68,3 +79,14 @@ def _exclude_tournament_field(self):
     def save_model(self, request, obj, form, change):
         self._process_tournament_field(obj, request)
         obj.save()
+
+    def formfield_for_foreignkey(self, db_field, request=None, **kwargs):
+        if db_field.name in self._fields_of_filter:
+            user = request.user
+            if not user.is_superuser:
+                if hasattr(user, 'scifight_user_profile'):
+                    tournament_of_user = user.scifight_user_profile.tournament
+                    kwargs["queryset"] = self._fields_of_filter[db_field.name].filter(
+                        tournament=tournament_of_user)
+
+        return super().formfield_for_foreignkey(db_field, request, **kwargs)
```

Comment:
I meant that filtering logic is the same, it's only fields to filter which differ.